galaxy/tools/deps/mulled/mulled_build.py

```diff
@@ -0,0 +1,274 @@
+            if submeta is None:
+                submeta = {}
+            res = submeta.get(key)
+            if res is None:
+                res = default
+            return res
+
+from .util import quay_versions, split_tag
+
+DIRNAME = os.path.dirname(__file__)
+
+
+def natural_key(string_):
+    """See http://www.codinghorror.com/blog/archives/001018.html."""
+    return [int(s) if s.isdigit() else s for s in re.split(r'(\d+)', string_)]
+
+
+def get_tests(args, pkg_path):
+    """Extract test cases given a recipe's meta.yaml file."""
+    recipes_dir = args.recipes_dir
+
+    tests = ""
+    input_dir = os.path.dirname(os.path.join(recipes_dir, pkg_path))
+    recipe_meta = MetaData(input_dir)
+
+    tests_commands = recipe_meta.get_value('test/commands')
+    tests_imports = recipe_meta.get_value('test/imports')
+    requirements = recipe_meta.get_value('requirements/run')
+
+    if tests_imports or tests_commands:
+        if tests_commands:
+            tests = ' && '.join(tests_commands)
+        elif tests_imports and 'python' in requirements:
+            tests = ' && '.join('python -c "import %s"' % imp for imp in tests_imports)
+        elif tests_imports and ('perl' in requirements or 'perl-threaded' in requirements):
+            tests = ' && '.join('''perl -e "use %s;"''' % imp for imp in tests_imports)
+        tests = tests.replace('$R ', 'Rscript ')
+    else:
+        pass
+    return tests
+
+
+def get_pkg_name(args, pkg_path):
+    """Extract the package name from a given meta.yaml file."""
+    recipes_dir = args.recipes_dir
+
+    input_dir = os.path.dirname(os.path.join(recipes_dir, pkg_path))
+    recipe_meta = MetaData(input_dir)
+    return recipe_meta.get_value('package/name')
+
+
+def get_affected_packages(args):
+    """Return a list of all meta.yaml file that where modified/created recently.
+
+    Length of time to check for indicated by the ``hours`` parameter.
+    """
+    recipes_dir = args.recipes_dir
+    hours = args.diff_hours
+    cmd = """cd '%s' && git log --diff-filter=ACMRTUXB --name-only --pretty="" --since="%s hours ago" | grep -E '^recipes/.*/meta.yaml' | sort | uniq""" % (recipes_dir, hours)
+    pkg_list = check_output(cmd, shell=True)
+    ret = list()
+    for pkg in pkg_list.strip().split('\n'):
+        if pkg and os.path.exists(os.path.join( recipes_dir, pkg )):
+            ret.append( (get_pkg_name(args, pkg), get_tests(args, pkg)) )
+    return ret
+
+
+def check_output(cmd, shell=True):
+    return subprocess.check_output(cmd, shell=shell)
+
+
+def conda_versions(pkg_name, file_name):
+    """Return all conda version strings for a specified package name."""
+    j = json.load(open(file_name))
+    ret = list()
+    for pkg in j['packages'].values():
+        if pkg['name'] == pkg_name:
+            ret.append('%s--%s' % (pkg['version'], pkg['build']))
+    return ret
+
+
+def new_versions( quay, conda ):
+    """Calculate the versions that are in conda but not on quay.io."""
```

Comment:
Do you have a problem with these functions in Galaxy-lib? We can remove them - I can workaround this on my side. Or this can be in planemo?